fetch-in-react/my-app/src/User.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import UserCard from './UserCard';
+
+export default function User({ userId, onCancel }) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+  const [user, setUser] = useState();
+
+  /* your code here (hint: useEffect) */
+
+  useEffect(() => {
+    async function logData() {
+      try {
+        const response = await fetch(
+          `https://jsonplaceholder.typicode.com/users/${userId}`
+        );
+        if (!response.ok) throw new Error(`fetch Error ${response.status}`);
+        const jsonData = await response.json();
+        if (jsonData) {
```

Comment:
Is it possible to not get jsonData back? I'm not sure. If it were to happen, though, you would not set isLoading to false so the user would have no idea what happened.